vega_sim/environment/agent.py

```diff
@@ -47,14 +50,32 @@ def __init__(
             key_name:
                 str, optional, Name of key in wallet for agent to use. Defaults
                 to value in the environment variable "VEGA_DEFAULT_KEY_NAME".
+            market_params_update_freq:
+                optional, int, Get the latest market parameters every n steps.
+            network_params_update_freq:
+                optional, int, Get the latest network parameters every n steps.
+            network_params:
+                optional, List[str], List of network parameters to get.
         """
         super().__init__()
         self.wallet_name = wallet_name
         self.wallet_pass = wallet_pass
         self.key_name = key_name
 
-    def step(self, vega: VegaService):
-        pass
+        self.step_count = 0
+        self.market_params_update_freq = market_params_update_freq
+        self.network_params_update_freq = network_params_update_freq
+
+        self.market_params = None
+        self.network_params = (
+            dict.fromkeys(network_params, None) if network_params is not None else None
+        )
```

Comment:
This feels perhaps a little too specific to our immediate use case to me. I'd probably stick to it just being a generic "state_update_freq" and then have individual agents update network params within that if they feel like it. But maybe params updates are general enough they deserve to have their own thing I'm not sure.

In any case I would certainly aggregate into one function at least, don't think we need to update them at different frequencies.

Reply:
I think you might be right, I was going for a super general function which could be used by any agent, but it probably overcomplicates things. 

Will move things around to match your suggestion  😄